### PR TITLE
Host orchestrator reverse proxies to operator

### PR DESCRIPTION
**Pull request by jemoreira**
Eliminates the redundancy between host-orchestrator and operator by having the signaling server features only in the operator.
Allows having and executing both binaries simultaneously, which is required for the orchestration UI to support local use cases.
The host orchestrator service now starts both the host_orchestrator and the operator daemons, with the operator listening only on localhost:2080 and the orchestrator on its regular ports.

### CODE REVIEW

File: frontend/debian/cuttlefish-user.cuttlefish-operator.default [hunk 1]
```diff
@@ -5,8 +5,8 @@
 # operator_http_port=
 #
 # The port on which the operator server should listen on for HTTPS.
-# Defaults to 1443.
-# operator_https_port=
+# Server over http only when not provided.
```
Comment (ser-io) [inline frontend/debian/cuttlefish-user.cuttlefish-operator.default:12]: nit: Serve?

File: frontend/src/host_orchestrator/main.go [hunk 2]
```diff
@@ -27,20 +27,15 @@ import (
...
 
 const (
-	DefaultSocketPath     = "/run/cuttlefish/operator"
-	DefaultHttpPort       = "1080"
-	DefaultTLSCertDir     = "/etc/cuttlefish-common/host_orchestrator/cert"
-	DefaultStaticFilesDir = "static"    // relative path
-	DefaultInterceptDir   = "intercept" // relative path
-	DefaultWebUIUrl       = ""
-
+	DefaultSocketPath               = "/run/cuttlefish/operator"
```
Comment (ser-io) [inline frontend/src/host_orchestrator/main.go:36]: it seems this is not used anymore.

File: frontend/src/host_orchestrator/main.go [hunk 3]
```diff
@@ -153,20 +130,22 @@ func main() {
...
+		log.Fatal("The host orchestrator requires access to the operator")
 	}
+	operatorURL, err := url.Parse(fmt.Sprintf("http://127.0.0.1:%s", operatorPort))
+	if err != nil {
+		log.Fatalf("Invalid operator port (%s): %v", operatorPort, err)
+	}
+	proxy := httputil.NewSingleHostReverseProxy(operatorURL)
+	// Defer to the operator for every route not covered by the orchestrator. That
+	// includes web UI and other static files.
+	r.PathPrefix("/").Handler(proxy)
```
Comment (ser-io) [inline frontend/src/host_orchestrator/main.go:144]: suggest to have this line and `imController.AddRoutes(r)` close to each other, this way we have a very clear location where all the routes are added.

**Review by ser-io — CHANGES_REQUESTED** [review]

File: frontend/debian/cuttlefish-orchestration.cuttlefish-host_orchestrator.init [hunk 4]
```diff
@@ -39,9 +39,11 @@ orchestrator_tls_cert_dir=${orchestrator_tls_cert_dir:-"/etc/cuttlefish-common/h
 orchestrator_cvd_artifacts_dir=${orchestrator_cvd_artifacts_dir:-"/var/lib/cuttlefish-common"}
 
 RUN_DIR="/run/cuttlefish"
-ASSET_DIR="/usr/share/cuttlefish-common/host_orchestrator"
-DAEMON="/usr/lib/cuttlefish-common/bin/host_orchestrator"
-PIDFILE="${RUN_DIR}"/host_orchestrator.pid
+ORCHESTRATOR_EXE="/usr/lib/cuttlefish-common/bin/host_orchestrator"
```
[the remark this answers is not in the document]
Comment (jemoreira) [reply frontend/debian/cuttlefish-orchestration.cuttlefish-host_orchestrator.init]: Done

File: frontend/debian/cuttlefish-user.cuttlefish-operator.default (hunk 1 above)
Comment (jemoreira) [reply frontend/debian/cuttlefish-user.cuttlefish-operator.default]: Done

File: frontend/src/host_orchestrator/main.go (hunk 2 above)
Comment (jemoreira) [reply frontend/src/host_orchestrator/main.go]: Done

File: frontend/src/host_orchestrator/main.go (hunk 3 above)
Comment (jemoreira) [reply frontend/src/host_orchestrator/main.go]: Done

**Review by ser-io — APPROVED** [review]

**Pull request merged**